### PR TITLE
Big boom, lots of changes - schema revision, multiple definitions 

### PR DESCRIPTION
**Pull request by josefbenjac**
### OpenApiPlugin 

- configuration `swagger` was changed to `swaggerUi`
- configuration `info` was changed to `definition`, now you can fill whole openApi specification

```neon
Apitte\OpenApi\DI\OpenApiPlugin:
	definition:
		openapi: "3.0.0"
		info:
			title: Swagger Petstore
			license:
				name: MIT
			version: 1.0.0
		servers:
			- url: http://petstore.swagger.io/v1

		paths: ...	
```

- you can also define standalone files using `files` configuration

```neon
Apitte\OpenApi\DI\OpenApiPlugin:
	files:
		- %appDir%/openApi/petstore.neon
		- %appDir%/openApi/petstore.yaml
		- %appDir%/openApi/petstore.json
```

- these files will be merged with other definitions

### Definitions

- now we support multiple definitions of Open Api schema
	- `BaseDefinition` - contains required fields, so you don't need specify
	- `AnnotationDefiniton` - core OpenApi annotations (old OpenApiService)
	- `ArrayDefinition` - config `definition` section
	- `FileDefintion[]` - config `files` section

- if you need add own definition, you can use `IDefinition` interface
- if you need change order, you can use or extend `SchemaBuilder`

### SchemaBuilder, ISchemaBuilder 

- `OpenApiService` was replaced by `SchemaBuilder` and `AnnotationDefinition`
- it has now interface, so you can use `ISchemaBuilder->build()` instead of `OpenApiService->createSchema()`

### Schema objects 

- most of Schema objects have `fromArray`, `toArray` methods

### Tests

- `fromArray`, `toArray` - on official examples

## TODO:

- [x] fix schema objects on official examples
- [x] plugin - files should support `appDir`
- [x] AnnotationDefinition
- [x] tests - add all examples
- [x] fix codestyle
- [x] docs

### Nice-to-have

- [x] BaseDefinition - required fields of openApi schema to fill AnnotationDefinition
- [x] JsonDefinition - load Json files
- [x] YamlDefinition - load yaml files

### CODE REVIEW

**Comment by f3l1x** [comment]
👍

File: composer.json [hunk 1]
```diff
@@ -22,11 +22,17 @@
   ],
   "require": {
     "php": ">= 7.1",
-    "apitte/core": "^0.4.0",
+    "apitte/core": "dev-master as 0.4.0",
```
Comment (mabar) [inline composer.json]: ^0.5.0 please, it's alias for current dev-master

File: src/Schema/Server.php [hunk 2]
```diff
@@ -12,27 +12,82 @@ class Server implements IOpenApiObject
...
 	{
 		$this->description = $description;
 	}
 
+	public function setVariable(string $key, ServerVariable $variable): void
+	{
+		$this->variables[$key] = $variable;
+	}
+
+	public function unsetVariable(string $key): void
```
Comment (mabar) [inline src/Schema/Server.php:65]: `addVariable()` and `removeVariable()`

File: src/SchemaDefinition/CoreDefinition.php [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	 */
+	protected function createPaths(): array
+	{
+		$paths = [];
+		foreach ($this->getEndpoints() as $endpoint) {
+			$operations = [];
+			foreach ($endpoint->getMethods() as $method) {
+				$operations[strtolower($method)] = $this->createOperation($endpoint);
+			}
+			$paths[$endpoint->getMask() ?? ''] = $operations;
```
Comment (mabar) [inline src/SchemaDefinition/CoreDefinition.php]: Should work with null too http://sandbox.onlinephpfunctions.com/code/6aa7b4a710cf2ee3b8820357ebeb58b27a2981b0

**Comment by mabar** [comment]
Great work. I like it.

**Pull request merged**